crates/uv-installer/src/satisfies.rs

```diff
@@ -130,10 +131,12 @@ impl RequirementSatisfaction {
                     return Ok(Self::Mismatch);
                 }
 
-                if installed_reference.as_deref() != requested_reference.as_str() {
+                if installed_reference.as_deref() != requested_reference.as_str()
+                    && installed_reference != &requested_precise.map(|git_sha| git_sha.to_string())
```

Comment:
E.g. the requested reference is the default branch, but we resolved and therefore know that the installed is at the same commit as the default branch is at.